presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -157,11 +157,26 @@ public SemiTransactionalHiveMetastore(
     }
 
     public synchronized Optional<Table> getTable(String databaseName, String tableName)
+    {
+        return getTable(databaseName, tableName, Optional.empty());
+    }
+
+    public synchronized Optional<Table> getTable(String databaseName, String tableName, Optional<ConnectorSession> session)
     {
         checkReadable();
         Action<TableAndMore> tableAction = tableActions.get(new SchemaTableName(databaseName, tableName));
         if (tableAction == null) {
-            return delegate.getTable(databaseName, tableName);
+            // TODO do it better – probably add semantic type instead of remove in case of leaks
+            Optional<Table> table = delegate.getTable(databaseName, tableName);
+            // TODO cannot import HiveSessionProperties class here to do this check because of circular dependency on presto-hive
+            if (table.isPresent() && session.isPresent() && !session.get().getProperty("semantic_types_enabled", Boolean.class)) {
```

Comment:
How strongly do we feel about keeping this settings as a session property? Do we think a configuration property is not flexible enough? Should we allow users change type semantics like that with a session property?

Reply:
The reason I feel pretty strongly about this is to facilitate deployment – a per-query opt-in limits the blast radius of these kind of changes.
Consider the following scenario: somebody changes the type of a VARCHAR column to an enum on an existing table – now all the queries that query this table can potentially break as a result. Whereas with a per-query flag we have more granular control of how we roll this out.
Wdyt?

Reply:
Can we just use `HiveSessionProperties#isSemanticTypesEnabled`? (similar to how other session properties get fetched)? -- we avoid hard code session property string like `semantic_types_enabled`